l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py

```diff
@@ -0,0 +1,319 @@
+            tmp_d = tempfile.mkdtemp()
+            tmp_key = self._create_tmp_file(key.private_key_crypted, tmp_d)[0]
+            tmp_f = self._create_tmp_file(self.attachment_id.datas, tmp_d)[0]
+            old_path_f = os.path.join(tmp_d, tmp_f.name)
+            new_path_f = os.path.join(tmp_d, ''.join(self.filename.split(':')))
+            shutil.move(old_path_f, new_path_f)
+
+            # upload to sftp
+            with pysftp.Connection(self.fds_account.hostname,
+                                   username=self.fds_account.username,
+                                   private_key=tmp_key.name,
+                                   private_key_pass=key_pass) as sftp:
+
+                with sftp.cd(self.fds_account_directory.name):
+                    sftp.put(new_path_f)
+                    _logger.info("[OK] upload file (%s) ", (self.filename))
+
+            # change to initial name file (mandatory because of the close)
+            shutil.move(new_path_f, old_path_f)
+            self._state_confirm_on()
+            self._add2historical()
+        except Exception as e:
+            self.attachment_id.unlink()
+            _logger.error("Unable to connect to the sftp: %s", e)
+            raise exceptions.Warning('Unable to connect to the sftp')
+
+        finally:
+            try:
+                tmp_key.close()
+            except:
+                _logger.error("remove tmp_key file failed")
+            try:
+                tmp_f.close()
+            except:
+                _logger.error("remove tmp_f file failed")
+            try:
+                shutil.rmtree(tmp_d)
+            except:
+                _logger.error("remove tmp directory failed")
+
+        return self._do_populate_tasks()
+
+    @api.multi
+    def back_button(self):
+        ''' Go back to the finish view.
+            Called by pressing back button.
+
+            :returns action: configuration for the next wizard's view
+        '''
+        self.ensure_one()
+        self._state_finish_on()
+        return self._do_populate_tasks()
+
+    @api.multi
+    def cancel_button(self):
+        ''' Remove the attachment create.
+            Called by pressing cancel button.
+
+            :returns: None
+        '''
+        self.ensure_one()
+        self.attachment_id.unlink()
+
+    ##############################
+    #          function          #
+    ##############################
+    @api.multi
+    def _get_sftp_key(self):
+        ''' private function that get the sftp key need for connection
+            with the server.
+
+            :returns (str key, str key_pass):
+            :riases Warning: if no key found
+        '''
+        fds_authentication_key_obj = self.env['fds.authentication.keys']
+        key = fds_authentication_key_obj.search([
+            ['user_id', '=', self.env.user.id],
+            ['fds_account_id', '=', self.fds_account.id]])
+
+        if not key:
+            raise exceptions.Warning('You don\'t have key')

```

Comment:
You could use double quote for all error message to avoid this kind of escaping.